ar4_realbot_bringup/launch/main.launch.py

```diff
@@ -30,48 +28,48 @@
 # OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+"""launch file for integrating Gazebo with MoveIt for the AR4 robot."""
 
-import os
-from ament_index_python.packages import get_package_share_directory
 from launch import LaunchDescription
-from launch.actions import DeclareLaunchArgument, IncludeLaunchDescription
+from launch_ros.actions import SetParameter
+from launch_ros.substitutions import FindPackageShare
+from launch.actions import IncludeLaunchDescription
 from launch.launch_description_sources import PythonLaunchDescriptionSource
-from launch.substitutions import LaunchConfiguration
-
-
-pkg_ar4_gazebo = get_package_share_directory('ar4_gazebo')
-pkg_ros_gz_sim = get_package_share_directory('ros_gz_sim')
+from launch.substitutions import PathJoinSubstitution
 
 
 def generate_launch_description():
-    use_sim_time = LaunchConfiguration('use_sim_time')
-    declare_use_sim_time_cmd = DeclareLaunchArgument(
-        'use_sim_time',
-        default_value='true',
-        description='Use simulation (Gazebo) clock if true',
-    )
+    """Launch the AR4 robot in Gazebo and MoveIt."""
+    use_sim_time_param = SetParameter(name="use_sim_time", value=True)
 
-    # Launch ar4 robot
-    ar4_launch = IncludeLaunchDescription(
+    common_stack_include = IncludeLaunchDescription(
         PythonLaunchDescriptionSource(
-            os.path.join(pkg_ar4_gazebo, 'launch', 'spawn_ar4.launch.py')
-        ),
-        launch_arguments={
-            'use_sim_time': use_sim_time,
-        }.items(),
+            PathJoinSubstitution(
+                [
+                    FindPackageShare("ar4_common"),
+                    "launch",
+                    "main.launch.py",
+                ]
+            )
+        )
     )
 
-    # Launch gazebo empty world
-    gazebo = IncludeLaunchDescription(
+    real_hardware_include = IncludeLaunchDescription(
         PythonLaunchDescriptionSource(
-            os.path.join(pkg_ros_gz_sim, 'launch', 'gz_sim.launch.py')
-        ),
-        launch_arguments={'gz_args': '-r empty.sdf'}.items(),
+            PathJoinSubstitution(
+                [
+                    FindPackageShare("ar4_hardware_interface"),
+                    "launch",
+                    "main.launch.py",
+                ]
+            )
+        )
     )
 
-    ld = LaunchDescription()
-    ld.add_action(declare_use_sim_time_cmd)
-    ld.add_action(ar4_launch)
-    ld.add_action(gazebo)
-
-    return ld
+    return LaunchDescription(
+        [
+            use_sim_time_param,
```

Comment:
Is `sim_time` true in the real robot?

Reply:
Good catch, fixed. I started from a copy of the other package and this was left here by mistake.